src/plugins/data_table/data_table.js

```diff
@@ -2,30 +2,249 @@ var LocalDataTable = (function() {
+        <button class="btn btn-primary btn-sm btn-clear" name="button" type="submit" title="">Clear</button>\
+      </div>\
+    ', null, DEFAULT_JST_DELIMS),
+
+    events: {
+      "click .toggle-filter-button": "_onToggleClick",
+      "click .btn-filter": "_onFilterClick",
+      "click .btn-clear": "_onClearClick"
+    },
+
+    initialize: function(options) {
+      this.filter = options.column.filter;
+      this.attr = options.column.attr;
+      this.title = options.column.title;
+      this.table = options.table;
+      this.head = options.head;
+      this.filterButtonClass = "filterInactive";
+
+      // decide which filter view based on type, here
+      var filterMenu = null;
+      switch (this.filter.type) {
+        case 'string':
+          filterMenu = new StringFilterMenu({column: options.column, parentView: this});
+          break;
+        case 'numeric':
+          filterMenu = new NumericFilterMenu({column: options.column, parentView: this});
+          break;
+        case 'list':
+          filterMenu = new ListFilterMenu({column: options.column, parentView: this});
+          break;
+      }
+
+      // add as a child (backdraft thing just to keep bookkeeping on subviews)
+      this.child("filter-menu", filterMenu);
+    },
+
+    render: function() {
+      this.$el.html(this.template({
+        filterButtonClass : this.filterButtonClass
+      }));
+
+      $(".filterMenu", this.$el).prepend(this.child("filter-menu").render().$el);
+      return this;
+    },
+
+    _toggleIcon: function(enabled) {
+      var icon = $("span", this.$el);
+      icon.removeClass("filterActive");
+      icon.removeClass("filterInactive");
+      if (enabled) {
+        icon.addClass("filterActive");
+      } else {
+        icon.addClass("filterInactive");
+      }
+    },
+
+    _onToggleClick: function(event) {
+      var table = this.table;
+      event.stopImmediatePropagation();
+      var currentFilterMenu = $('.filterMenu', this.$el);
+      if ((table.activeFilterMenu) && (table.activeFilterMenu.is(currentFilterMenu))) {
+        table.activeFilterMenu.slideUp(100);
+        table.activeFilterMenu = null;
+      } else if (table.activeFilterMenu) {
+        table.activeFilterMenu.slideUp(100, function () {
+          table.activeFilterMenu = currentFilterMenu;
+          table.activeFilterMenu.slideDown(200);
+        });
+      } else {
+        table.activeFilterMenu = currentFilterMenu;
+        table.activeFilterMenu.slideDown(200);
+      }
+    },
+
+    _onFilterClick: function() {
+      $("input[type=text]", this.head).trigger("change");
+      this.table.dataTable._fnAjaxUpdate();
+    },
+
+    _onClearClick: function() {
+      $("input[type=text]", this.head).val("");
+      $("input[type=checkbox]", this.head).attr("checked", false);
+      $("input", this.head).trigger("change");
+      this.table.dataTable._fnAjaxUpdate();
+    }
+  });
+
```

Comment:
is it easy to put all the above in a separate .js file like we discussed? I hope we can follow the patterns set by other files like column_manager.js